src/sif/dynamiccost.cc

```diff
@@ -261,18 +270,28 @@ float DynamicCost::GetModeFactor() {
 }
 
 // Gets the hierarchy limits.
-std::vector<HierarchyLimits>& DynamicCost::GetHierarchyLimits() {
+std::vector<HierarchyLimits> DynamicCost::GetHierarchyLimits() {
   return hierarchy_limits_;
 }
 
+// Gets mutable hierarchy limits.
```

Comment:
This is now necessary, because the thor worker managing the hierarchy limits needs a copy of the originally passed hierarchy limits. Since we have different defaults for the different routing algorithms, and we may use them interchangeably for any given location pair, we always need two versions: one for bidirectional and one for timedep forward/reverse.